File: Python/project-euler/problems/problem_020.py
from utils.code_runner import execute_code


def problem():
    n = 100
    factorial = 1
    total_sum = 0
    # n! means n × (n − 1) × ... × 3 × 2 × 1
    # Get the product of all positive integers less than or equal to n
    for current_factorial in range(n, 0, -1):
        factorial *= current_factorial
    # Put the factorial in an int array and sum it up
    for digit in [int(x) for x in str(factorial)]:
        total_sum += digit

    return total_sum


if __name__ == '__main__':
    execute_code(problem)

File: Java/src/main/java/nl/davydehaas/projecteuler/Problem020.java
package nl.davydehaas.projecteuler;

import static nl.davydehaas.projecteuler.util.SolutionTimer.timeSolution;

import java.math.BigInteger;
import java.util.Arrays;

public class Problem020 {

    void main() {
        timeSolution(Problem020::solve);
    }

    static int solve() {
        int n = 100;
        BigInteger factorial = BigInteger.ONE;
        // n! means n * (n - 1) * ... 3 * 2 * 1.
        // Get the product of all positive integers less than, or equal to, n.
        for (int currentFactorial = n; currentFactorial > 0; currentFactorial--) {
            factorial = factorial.multiply(BigInteger.valueOf(currentFactorial));
        }

        // Find the sum of the digits in the number.
        return Arrays.stream(factorial.toString().split(""))
                .mapToInt(Integer::parseInt)
                .sum();
    }
}
